Upstream status: pfSense only, PR opened 
FreeBSD PR: usb/173503: Add IDs for Huawei K3770 and K4507

--- ./sys/dev/usb/serial/u3g.c	2012-03-06 14:37:00.000000000 +0100
+++ ./sys/dev/usb/serial/u3g.c.new	2012-06-03 18:19:07.000000000 +0200
@@ -290,6 +290,8 @@
 	U3G_DEV(HUAWEI, E1820, U3GINIT_HUAWEISCSI),
 	U3G_DEV(HUAWEI, K3765, U3GINIT_HUAWEI),
 	U3G_DEV(HUAWEI, K3765_INIT, U3GINIT_HUAWEISCSI),
+	U3G_DEV(HUAWEI, K4505_INIT, U3GINIT_HUAWEISCSI),
+	U3G_DEV(HUAWEI, K4505, U3GINIT_HUAWEI),
 	U3G_DEV(HUAWEI, K3770, U3GINIT_HUAWEI),
 	U3G_DEV(HUAWEI, K3770_INIT, U3GINIT_HUAWEISCSI),
 	U3G_DEV(KYOCERA2, CDMA_MSM_K, 0),
--- sys/dev/usb/usbdevs	2012-06-03 18:26:29.000000000 +0200
+++ sys/dev/usb/usbdevs.new	2012-06-03 18:26:22.000000000 +0200
@@ -1877,6 +1877,8 @@
 product HUAWEI E1752		0x1446	3G modem
 product HUAWEI K3765		0x1465	3G modem
 product HUAWEI E1820		0x14ac	E1820 HSPA+ USB Slider
+product HUAWEI K4505_INIT	0x1521	K4505 Initial
+product HUAWEI K4505		0x1464	3G modem
 product HUAWEI K3770		0x14c9	3G modem
 product HUAWEI K3770_INIT	0x14d1	K3770 Initial
 product HUAWEI K3765_INIT	0x1520	K3765 Initial